docs/OCI_COMPLIANCE.md

```diff
@@ -0,0 +1,256 @@
+- SIGKILL: Force termination (kills proxy, processor may remain running)
+
+The firmware itself cannot receive signals - it runs on a separate processor without signal infrastructure.
+
+**Rationale**: Signals control the lifecycle management proxy, not the firmware. The firmware is controlled by writing to sysfs (`state` file).
+
+### 12. Single Container per Processor Limitation
+
+**Standard OCI**: Runtimes can create and run multiple independent containers simultaneously.
+
+**Remoteproc Runtime**: Can only run **one container at a time per remote processor**. Each auxiliary processor can execute only one firmware image. The previous container must be stopped before a new one can be started.
+
+**Rationale**: Hardware constraint - auxiliary processors have single execution contexts and cannot run multiple firmware images concurrently.
+
+**Impact**: Container orchestrators (Kubernetes, Docker Swarm) must be aware of this 1:1 mapping between containers and processor resources.
+
+## Unique Remoteproc Runtime Features
+
+### Firmware Lifecycle Management
+
+The runtime implements a two-phase proxy process lifecycle unique to remoteproc:
+
+**Phase 1**: Wait for start signal
+
+- Proxy blocks waiting for SIGUSR1 or termination signal
+- Allows separation of container creation and execution
+
+**Phase 2**: Monitor and maintain
+
+- Writes firmware filename to sysfs `firmware` attribute
+- Writes "start" to sysfs `state` attribute
+- Polls processor state
+- Exits if processor stops or crashes
+- Responds to graceful stop signals
+
+This design integrates with the Linux kernel's remoteproc framework expectations.
+
+### Processor State Mapping
+
+The runtime maps remoteproc kernel states to OCI states:
+
+| Remoteproc State          | OCI State | Description                     |
+| ------------------------- | --------- | ------------------------------- |
+| offline                   | created   | Firmware loaded but not started |
+| running                   | running   | Processor executing firmware    |
+| offline/suspended/crashed | stopped   | Processor not executing         |
+
+### Firmware Storage
+
+Firmware is copied to `/lib/firmware/` with a unique name:
+
+```
+<original-name>-<timestamp>-<random-suffix>
+```
+
+This prevents conflicts when multiple containers use the same firmware base name and enables cleanup on container deletion.
+
+### Annotations
+
+Required annotation in config.json ([OCI Config Spec - Annotations](https://github.com/opencontainers/runtime-spec/blob/main/config.md#annotations)):
+
+```json
+{
+  "annotations": {
+    "remoteproc.name": "imx-dsp-rproc"
+  }
+}
+```
+
+The runtime adds state annotations:
+
+- `remoteproc.resolved-path`: Full sysfs device path
+- `remoteproc.firmware`: Stored firmware filename in `/lib/firmware/`
+
+## References
+
+- [OCI Runtime Specification](https://github.com/opencontainers/runtime-spec)
+- [Linux Remoteproc Framework](https://docs.kernel.org/staging/remoteproc.html)
```

Comment:
In the rest of the document this is generally not capitalised. This is true also in the linked docs.
```suggestion
- [Linux remoteproc framework](https://docs.kernel.org/staging/remoteproc.html)
```